pkg/dns/dns.go

```diff
@@ -87,3 +89,9 @@ func (*FakeProvider) ProviderSpecific() ProviderSpecificLabels {
 		HealthCheckID: "fake/health-check-id",
 	}
 }
+
+func SanitizeErrormessage(err error) error {
+	regexp := regexp.MustCompile(`request id: [^\s]+`)
+	sanitizedErr := regexp.ReplaceAllString(err.Error(), "")
+	return errors.New(sanitizedErr)
+}
```

Comment:
I'd rename this to `SanitizeErrorMessage` or maybe even `SanitizeError`.

You might also want to consider adding a basic test for it, nothing major, just a couple of examples of what we expect in and the expected output.  I'd also add a [godoc](https://go.dev/blog/godoc) comment to the function explaining what it does, for example:

```
// SanitizeErrormessage removes request specific data from error messages in order to make them consistent across multiple similar requests to the provider.  e.g AWS SDK Request ids `request id: 051c860b-9b30-4c19-be1a-1280c3e9fdc4`
```